src/main/java/com/codingchili/Controller/Website.java

```diff
@@ -127,6 +127,10 @@ private void setRouterAPI(Router router) {
 
             if (iterator.hasNext()) {
                 MultiMap params = context.request().params();
+                if (Configuration.isIndexLocked()) {
+                	params.add(INDEX, Configuration.getDefaultIndex());

```

Comment:
When using `params#add` on the MultiMap it's possible to end up with multiple params for the INDEX key, if a client acts maliciously and sets this param in some other way. Using `params#set` instead, would remove any existing keys before insertion. I don't think we should be modifying the request params directly here, it would be more elegant to convert it to an ImportEvent here instead.

For example

```java
ImportEvent event = ImportEvent.fromParams(params);
                
if (Configuration.isIndexLocked()) {
    event.setIndex(Configuration.getDefaultIndex());
}
```
And then just change the parse call to accept an ImportEvent instead of a MultiMap.